app/javascript/src/js/darkModeBeta.js

```diff
@@ -0,0 +1,71 @@
+/* eslint-disable no-use-before-define */
+function enableDarkMode() {
+  const navbar = document.querySelector('.navbar-nav');
+  const userSelection = checkStorage();
+  const darkModeButton = createIcon(userSelection);
+  navbar.insertBefore(darkModeButton, navbar.lastElementChild);
+}
+
+function checkStorage() {
+  const darkModeOption = localStorage.getItem('topDarkMode');
+  if (darkModeOption === null) {
+    localStorage.setItem('topDarkMode', JSON.stringify(false));
+    return false;
+  }
+  return JSON.parse(darkModeOption);
+}
+
+async function toggleOption() {
+  const isDarkModeEnabled = checkStorage();
+  if (isDarkModeEnabled) {
+    document.querySelector('#darkModeStyles').remove();
+  } else {
+    const result = await fetch('https://cdn.jsdelivr.net/gh/TheOdinProject/top-dark-theme@master/darkMode.css');
+    const css = await result.text();
+    const style = document.createElement('style');
+    style.innerText = css;
+    style.id = 'darkModeStyles';
+    document.querySelector('head').appendChild(style);
+  }
+  localStorage.setItem('topDarkMode', JSON.stringify(!isDarkModeEnabled));
+  swapIcons(!isDarkModeEnabled);
+}
+
+function createIcon(darkModeEnabled) {
+  let darkModeButton;
+  if (darkModeEnabled) {
+    // Sun icon
+    darkModeButton = document.createRange().createContextualFragment(
+      `<li class="nav-item nav-link" id="dark-mode-item" style="flex-direction: row; justify-content: center; padding-right:0">
```

Comment:
Would it be possible to have these as static elements on the navbar instead of dynamically created elements? it would be safer in case something is added or changed which could break where these buttons are positioned.

Reply:
Thanks I'll check them out!